src/shared/handlers/onMessageEventHandler.js

```diff
@@ -0,0 +1,33 @@
+import {
+  sendRawLog,
+  setToStorage,
+  handleUserRegistration,
+  handleUserDeletion,
+  postMessageToFirstActiveTab,
+} from "..";
+
+export const onMessageEventHandler = async (request) => {
+  if (request.action === "sendRawLog") {
+    const { action, ...payload } = request;
+    sendRawLog(payload);
+  } else if (request.registerWTMUser) {
+    const { registerWTMUser, ...payload } = request;
+    await handleUserRegistration(payload, async (response) => {
+      if (process.env.BROWSER === "firefox") {
```

Comment:
Again, this looks like it should be a switch.

In the old days, we used to say "you should detect features, not browsers". Showing my age there 😬

Reply:
Very interesting. How do you suggest we go about detecting features?
If we have a switch statement, we most likely have to separate that into a script that will return maybe an object? Something like:

```
switch(BROWSER) {
     case "chrome":
         return {webpack_entry: {...}, onMessageEventHandler: ()=>{} };
}
```

This looks a bit unusual to me, a few unrelated (somehow related by having different browsers) attributes are put together in an object.

Reply:
I'm not sure how we detect features. If chrome.serviceworker exists? We don't have to do that, I know we've got targetted build params. Just a thought. **

** If we really wanted to do this, I'm not saying we should, but if we did, we could use https://modernizr.com

Reply:
I know switches are more verbose, so people don't use them much. But they're so damn explicit, and represent list selection so well, I think they're cool.

```js
let entry;
switch (process.env.BROWSER) {
  case 'chrome':
  case 'edge':
    entry = {
      worker: __dirname + "/../daemon/background/worker.js",
      initOverload: __dirname + "/../daemon/collector/overload/initOverload.js",
      overload: __dirname + "/../daemon/collector/overload/overload.js",
    }
    break;

  case 'firefox':
    entry = {
      index: __dirname + "/../daemon/index.js",
      overload: __dirname + "/../daemon/collector/overload/overload.js",
    }
    break;

  default:
    throw Error('process.env.BROWSER must be defined')   
}
```